### PR TITLE
Simplify binary-search in FindRunIndex

### PR DESCRIPTION
**Pull request by felipecrv**
This is my first PR to DuckDB: it contains a simplification of the binary search in `arrow_conversion.cpp` — I don't have any reason to believe the original version has bugs, but the new implementation is easier to analyze.

I tested it with `BUILD_PYTHON=1 make -C build/debug install && pytest ./tools/pythonpkg/tests/fast/arrow/test_arrow_run_end_encoding.py`

### CODE REVIEW

**Comment by pdet** [comment]
Valeu pela contribuição, Felipe! :-)

@szarnyasg Since that's @Tishj 's code, I think he can provide a more in-depth review, so I changed the review request to him if that's ok!

File: src/function/table/arrow_conversion.cpp [hunk 1]
```diff
@@ -372,30 +372,31 @@ static void IntervalConversionMonthDayNanos(Vector &vector, ArrowArray &array, c
...
+	//    3, 4 -> 1
+	//       5 -> 2
+	// 6, 7 .. -> 3 (3 == count [not found])
 	idx_t begin = 0;
-	idx_t end = count - 1;
+	idx_t end = count;
 	while (begin < end) {
-		idx_t middle = static_cast<idx_t>(std::floor((begin + end) / 2));
+		idx_t middle = (begin + end) / 2;
+		// begin < end impies middle < end
```
Comment (Tishj) [inline src/function/table/arrow_conversion.cpp:390]: implies*

File: src/function/table/arrow_conversion.cpp [hunk 2]
```diff
@@ -372,30 +372,31 @@ static void IntervalConversionMonthDayNanos(Vector &vector, ArrowArray &array, c
...
-				return middle;
-			}
-			if (offset >= static_cast<idx_t>(run_ends[middle - 1])) {
-				// For example [0, 0, 0, 1, 1, 2]
-				// encoded as run_ends: [3, 5, 6]
-				// 3 (run_ends[0]) >= offset < 5 (run_ends[1])
-				return middle;
-			}
-			end = middle - 1;
+			// run_ends[middle] > offset, so keep searching in [begin, middle)
```
Comment (Tishj) [inline src/function/table/arrow_conversion.cpp:395]: >=

**Review by Tishj — CHANGES_REQUESTED** [review]
Thanks, just left some comments on the comments

File: src/function/table/arrow_conversion.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (felipecrv) [reply src/function/table/arrow_conversion.cpp]: Sure.

**Review by Tishj — APPROVED** [review]
Thanks, looks much simpler 👍

**Pull request merged**